Creating-Objects.md

```diff
@@ -1,12 +1,105 @@
-
-Let's suppose you're creating a new object called `MyObject` (in practice you should use a more descriptive name).
-  1. In `extensions/objects/models/objects.py` create a new object, with either `BaseObject` or one of the other existing objects as its superobject. The class should have the following attributes:
-       * `description`: A string describing the object.
-       * (optional) `edit_html_filename`: The name of the file in (3) below, if it exists.
-       * (optional) `edit_js_filename`: The name of the file in (4) below, if it exists.
-       * `default_value`: This is needed for any object that is used as an input parameter to a rule. It represents the default value of that parameter.
-       * `normalize`: A function that is given an instance of the object in question and is responsible for validating and normalizing it. The supplied instance is a native Python object.
-  1. In `extensions/objects/models/objects_test.py` add new tests to the `ObjectNormalizationUnitTests` class; these will be run automatically with the other tests.
-  1. (optional) In `extensions/objects/templates` add a file `my_object_editor.html` specifying an interface in which objects of this type can be submitted. This is only necessary if editors will be entering parameters that use this object type.
-  1. (optional) In `extensions/objects/templates`, create `myObjectEditor.js`, the JavaScript directive associated with `my_object_editor.html`. You should base this (and the HTML file) on one of the pre-existing ones.
+## Table of Contents
+
+* [Introduction](#introduction)
+* [How objects are defined](#how-objects-are-defined)
+  * [Python classes](#python-classes)
+  * [Angular components](#angular-components)
+* [Create a new object](#create-a-new-object)
+* [Using objects](#using-objects)
+
+## Introduction
+
+When collecting user input, we often define the kind of input we expect using [[schemas|Schemas]]. These schemas can describe arbitrarily nested collections of base types we call _objects_. For example, consider this schema:
+
+```json
+{
+  "type": "dict",
+  "properties": [
+    {
+      "name": "version",
+      "schema": {
+        "type": "int"
+      }
+    }
+  ]
+}
+```
+
+The schema would match a dictionary like this:
+
+```json
+{"version": 34}
+```
+
+The `dict` type is a collection, but the `int` type is an object. We define these objects as classes in [`extensions/objects/models/objects.py`](https://github.com/oppia/oppia/blob/develop/extensions/objects/models/objects.py) and components in [`extensions.objects/templates`](https://github.com/oppia/oppia/blob/develop/extensions/objects/templates).
```

Comment:
I don't understand the first sentence here. What is a "collection" in this context?

Reply:
The "collection" here refers to the collections mentioned in the previous paragraph:

> These schemas can describe arbitrarily nested collections of base types. We call these base types _objects_.

I've re-worked the sentence to focus on the objects, since that's the important part.